amp/parse_test.go

```diff
@@ -25,7 +25,7 @@ func TestParseParams(t *testing.T) {
 		{
 			description: "All Fields",
 			query: "tag_id=anyTagID&account=anyAccount&curl=anyCurl&consent_string=anyConsent&debug=1&__amp_source_origin=anyOrigin" +
-				"&slot=anySlot&timeout=42&h=1&w=2&oh=3&ow=4&ms=10x11,12x13",
+				"&slot=anySlot&timeout=42&h=1&w=2&oh=3&ow=4&ms=10x11,12x13&targeting=%7B%22gam-key1%22%3A%22val1%22%2C%22gam-key2%22%3A%22val2%22%7D",
```

Comment:
I checked the test and it will require a lot of refactoring to implement this, including parts I didn't touch. 
I still think this is not necessary, because it shows exactly how input data will look like. This is useful for future reference. Expected result specified in the `expectedParams` one line below.

Reply:
Yeah, I see what you mean. We will have to store them as a map and then add them inside the for loop one by one. Do you think it might be helpful to add a comment with the decoded query string on the line above to make it easier to read? Have approved the PR anyway since it LGTM

Reply:
Thank you Mansi, I added comment.